File: branches/2.4.x multiple classmaps/Prototypes/Hibernate/src/org/iringtools/library/IDataLayer.java
package org.iringtools.library;

import java.util.List;
import org.iringtools.common.response.Response;
import org.iringtools.data.filter.DataFilter;

public interface IDataLayer {

	List<IDataObject> create(String objectType, List<String> identifiers);

	long getCount(String objectType, DataFilter filter);

	List<String> getIdentifiers(String objectType, DataFilter filter);

	List<IDataObject> get(String objectType, List<String> identifiers);

	List<IDataObject> get(String objectType, DataFilter filter, int pageSize,
			int startIndex);

	Response post(List<IDataObject> dataObjects);

	Response delete(String objectType, List<String> identifiers);

	Response delete(String objectType, DataFilter filter);

	// DataDictionary GetDictionary();

	List<IDataObject> getRelatedObjects(IDataObject dataObject,
			String relatedObjectType);
}

File: Prototypes/TDD/iRINGTools.Data/Model/Interfaces/IDataLayer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Linq;

namespace iRINGTools.Data
{
  public interface IDataObject
  {
    string ObjectType { get; }
    object GetPropertyValue(string propertyName);
    void SetPropertyValue(string propertyName, object value);
  }

  public interface IDataLayer
  {
    string Name { get; }

    IQueryable<IDataObject> DataObjects { get; }

    IList<IDataObject> Create(string objectType, IList<string> identifiers);

    IList<IDataObject> Get(string objectType, IList<string> identifiers);

    IList<IDataObject> Get(string objectType, DataFilter filter, int pageSize, int startIndex);

    long GetCount(string objectType, DataFilter filter);

    IList<string> GetIdentifiers(string objectType, DataFilter filter);
    
    Response Post(IList<IDataObject> dataObjects);

    Response Delete(string objectType, IList<string> identifiers);

    Response Delete(string objectType, DataFilter filter);
    
    Dictionary GetDictionary();

    IList<IDataObject> GetRelatedObjects(IDataObject dataObject, string relatedObjectType);
  }

  public interface IConfigurableDataLayer : IDataLayer
  {
    Response Configure(XElement configuration);
    XElement GetConfiguration();
  }

}
